Lib/typing.py

```diff
@@ -880,7 +880,7 @@ def __init__(self, arg, is_argument=True, module=None, *, is_class=False):
         # If we do `def f(*args: *Ts)`, then we'll have `arg = '*Ts'`.
         # Unfortunately, this isn't a valid expression on its own, so we
         # do the unpacking manually.
-        if arg[0] == '*':
+        if arg and arg[0] == '*':
```

Comment:
We could just change this to use `startswith()`, that'd be simpler and more readable.